### PR TITLE
Add basic intellij plugin

### PR DESCRIPTION
**Pull request by jacksonthea**

### CODE REVIEW

**Review by ggleyzer — COMMENTED** [review]
Please remove everything we don't know we neeed

File: .idea/misc.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
   <component name="FrameworkDetectionExcludesConfiguration">
     <file type="web" url="file://$PROJECT_DIR$" />
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_19" default="true" project-jdk-name="azul-18" project-jdk-type="JavaSDK" />
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_18" default="true" project-jdk-name="azul-18" project-jdk-type="JavaSDK" />
```
Comment (jacksonthea) [inline .idea/misc.xml:7]: I don't know why the jdk version would be different from the language level. How's that possible?

File: plugin_intellij/src/main/java/org/xtclang/sdk/language/EcstasyFileType.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
+package org.xtclang.sdk.language;
+
+import com.intellij.openapi.fileTypes.LanguageFileType;
```
Comment (ggleyzer) [inline plugin_intellij/src/main/java/org/xtclang/sdk/language/EcstasyFileType.java:3]: Please take at any .java in javatools project to seeur coding conventions (e.g. org.xvm.compiler.Compiler.java). We are usually separating the import packages with a new line, trying not to use wildcard imports unless it's more than 20 imports from the same package and comment anything that is not absolutely obvious.

**Review by ggleyzer — APPROVED** [review]

**Comment by jacksonthea** [comment]
Cleaned up the imports/packages. Want to take one more look?

**Pull request merged**